hal/api/critical.h

```diff
@@ -234,6 +287,14 @@ uint16_t core_util_atomic_incr_u16(uint16_t * valuePtr, uint16_t delta);
 uint32_t core_util_atomic_incr_u32(uint32_t * valuePtr, uint32_t delta);
 
 /**
+ * Atomic increment.
+ * @param  valuePtr Target memory location being incremented.
+ * @param  delta    The amount being incremented.
+ * @return          The new incremented value.
+ */
+void *core_util_atomic_incr(void **valuePtr, unsigned delta);
```

Comment:
why is `valuePtr` a pointer-pointer when its specializations (above) take `valuePtr` as a single-level pointer?